eventbusk/bus.py

```diff
@@ -31,6 +32,7 @@ class MyEvent(Event):
         foo: int
         bar: str
     """
+    event_id: str = field(default=False, init=False)
```

Comment:
Opinion: Consider using `default_factory=uuid.uuid4`. More: https://docs.python.org/3/library/dataclasses.html#dataclasses.field and you can convert to string while sending.

Reply:
Opinion: I feel `id` is better naming. Consider readability for `event.id` vs `event.event_id`

Reply:
I would be fine with `id` going forward. But not that adding this might cause other issues in existing events where id is used.

https://github.com/airbase/airbase-backend/blob/master/airbase_platform/receivers.py#L69-L70

I would assume the child classes can anyway override the base class definition? So we can figure out a transition for existing use cases if we need to do this.

cc @gman90 since @Airbase/vault will be one of the core users of kafka .

We can also get on a periodic call if needed, until we iron out some basics. I think there are a bunch of people that are starting out this quarter on the eventbus, and i've noticed some issues(which might or might not be something important).

Reply:
>So we can figure out a transition for existing use cases if we need to do this.

For https://github.com/airbase/airbase-backend/blob/master/airbase_platform/receivers.py#L69-L70
Either 
 1. Introduce a v2 event  OR 
 2. Add an additional `state_id` field (to replace `id`usage) at producer and add logic within consumer to handle this

Reply:
@GauthamGoli Anything is fine, as long as you ensure the existing topics don't break.

Reply:
The main thing about changing the base class id field is what happens to inflight events. At this point, the event class is the schema so changing that will cause an issue for inflight events. We are going to have to think about mirgrations

Reply:
After discussion with @sidmitra  and @gman90  we'd decided to go forward with `event_id` field to avoid refactor and migration effort. 

I've however updated `Event` base class to update type of `event_id` to `uuid.UUID`